kyo-reactive-streams/shared/src/main/scala/kyo/interop/reactive-streams/StreamSubscription.scala

```diff
@@ -0,0 +1,148 @@
+package kyo.interop.reactivestreams
+
+import StreamSubscription.*
+import kyo.*
+import kyo.interop.reactivestreams.*
+import kyo.kernel.ArrowEffect
+import kyo.kernel.Boundary
+import kyo.kernel.Safepoint
+import kyo.scheduler.IOPromise
+import kyo.scheduler.IOTask
+import org.reactivestreams.*
+
+final class StreamSubscription[V, Ctx] private[reactivestreams] (
+    private val stream: Stream[V, Ctx],
+    subscriber: Subscriber[? >: V]
+)(
+    using
+    allowance: AllowUnsafe,
+    boundary: Boundary[Ctx, IO],
+    tag: Tag[V],
+    frame: Frame
+) extends Subscription:
+
+    private val requestChannel = Channel.Unsafe.init[Long](Int.MaxValue, Access.SingleProducerSingleConsumer)
+
+    override def request(n: Long): Unit =
+        if n <= 0 then subscriber.onError(new IllegalArgumentException("non-positive subscription request"))
+        discard(requestChannel.offer(n))
+    end request
+
+    override def cancel(): Unit =
+        given Frame = Frame.internal
+        discard(requestChannel.close())
+    end cancel
+
+    private[reactivestreams] inline def subscribe: Unit < IO = IO(subscriber.onSubscribe(this))
+
+    private[reactivestreams] def poll: StreamFinishState < (Async & Poll[Chunk[V]]) =
```

Comment:
`Poll` is great, is make the logic much cleaner then the previous attemp

Reply:
cool!!